Documentation/container-lifecycle.md

```diff
@@ -0,0 +1,34 @@
+# Lifecycle of a container in rocket
+
+## Creation
+
+`rkt run` instantiates a container uuid and filesystem tree rooted at `/var/lib/rkt/containers/$uuid/.`, henceforth referred to as `$cdir`.
+
+An exclusive advisory lock is acquired on `$cdir` with a file descriptor associated with the process executing the stage1 /init.  This lock is automatically released when the /init process exits and the kernel closes all of the remaining open file descriptors.
+
+
+## Death / exit
+
+A container is considered exited if an advisory lock can be acquired on `$cdir`.  Upon exit of a container's /init process, the exclusive `$cdir` lock acquired at the beginning of `rkt run` becomes released by the kernel.
+
+## Garbage collection
+
+Exited containers are discarded in a common gc mark-and-sweep style by invoking the `rkt gc` command:
+
+### Mark
+
+All directories found in `/var/lib/rkt/containers` are tested for exited status by trying to acquire a shared lock on each directory.
+
+When a directory's lock cannot be acquired, the directory is skipped as it indicates the container is currently executing.
+
+For directories which the lock is successfully acquired, the directory is atomically renamed from `/var/lib/rkt/containers/$uuid` to `/var/lib/rkt/garbage/$uuid`.  The renaming effectively implements the "mark" operation.  The locks are immediately released, and operations like `rkt status` may occur simultaneous to `rkt gc`.
+
+### Sweep
+
+A side-effect of the rename operation responsible for moving a container from `/var/lib/rkt/containers` to `/var/lib/rkt/garbage` is an update to the directory's change time.  The sweep operation takes advantage of this in providing a grace-period before discarding exited containers.
+
+The change times of all directories found in `/var/lib/rkt/garbage` are compared against the current time.  Directories having sufficiently old change times are locked exclusively and recursively deleted.  If a lock acquisition fails, the directory is skipped.  Failed exclusive lock acquisitions may occur if the garbage container is currently being accessed via `rkt status`, for example.  The skipped containers will be revisited on a subsequent `rkt gc` invocation's sweep.
```

Comment:
Is this time configurable? If not, is it worth documenting the value you are using? Or is this just an implementation detail that most users don't care about?